task/net/src/server.rs

```diff
@@ -577,18 +710,26 @@ where
             return Err(ClientError::AccessViolation.fail());
         }
 
+        // Convert from absolute VID to an index in our VLAN array
         #[cfg(feature = "vlan")]
-        let vlan_index = {
-            // Convert from absolute VID to an index in our VLAN array
-            if !VLAN_RANGE.contains(&metadata.vid) {
-                return Err(ClientError::BadMessageContents.fail());
-            }
-            usize::from(metadata.vid - VLAN_RANGE.start)
+        let Some(vlan_index) = VLANS.iter().position(|v| v.vid == metadata.vid) else {
+            return Err(ClientError::BadMessageContents.fail());
```

Comment:
So it looks like in some cases, we _are_ reply-faulting in response to an invalid vlan tag? But in others (like monorail) we aren't

Reply:
Fixed in the Grand VLAN Refactoring; these functions now take an `enum VLanId`.